docs/getting-started/java.md

```diff
@@ -0,0 +1,254 @@
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Get Started with the OpenFeature Java SDK and Spring Boot
+
+## Introduction
+
+This walk-through teaches you the basics of using OpenFeature in Java on top of Spring Boot Web application.
+
+You'll learn how to:
+
+- Integrate OpenFeature Java SDK
+- Install and configure an OpenFeature provider
+- Perform basic feature flagging
+
+## Requirements
+
+This walk-through assumes that:
+
+- You have a basic knowledge of Java and Spring Boot
+- You have Java 8 or later
+- You have Docker installed and running on the host system
+
+## Walk-through
+
+### Step 1: Create a Spring Boot Web application
+
+To get started, visit [Spring Initializer](https://start.spring.io/). This website allows you to download a pre-configured
+Spring boot application. 
+
+For our use case, add **Spring Web** from dependencies section and set project metadata to your desired values.
+Download the application using **GENERATE** button. Extract the archive and open it in your favourite editor.
+
+### Step 2: Add dependencies
+
+Based on the selected build system for Spring Boot application, add OpenFeature Java SDK and flagd provider dependencies
+to existing dependency management configuration. 
+
+Below examples show dependencies for Maven and Gradle and contains the latest 
+versions at the time of writing this walk-through.
+
+<Tabs groupId="dependency">
+<TabItem value="maven" label="Maven">
+
+```xml
+<dependency>
+    <groupId>dev.openfeature</groupId>
+    <artifactId>sdk</artifactId>
+    <version>1.0.0</version>
+</dependency>
+<dependency>
+    <groupId>dev.openfeature.contrib.providers</groupId>
+    <artifactId>flagd</artifactId>
+    <version>0.5.1</version>
+</dependency>
+```
+
+</TabItem>
+<TabItem value="gradle" label="Gradle">
+
+```groovy
+implementation 'dev.openfeature:sdk:1.0.0'
+implementation 'dev.openfeature.contrib.providers:flagd:0.5.1'
+```
+</TabItem>
+</Tabs>
+
+### Step 3: Add code
+
+Our simple web server contains a single endpoint `/hello`. Let's add required codes to get this up and running with
+recommended best practices. 
+
+Please note that you may have your own package names and package naming in sample codes will require alterations.
+
+`OpenFeatureAPI` is the main access point to OpenFeature SDK. This object is designed to act as a singleton. Hence, let's 
+define a simple bean configuration class `OpenFeatureBeans`. This definition allows us to dependency inject `OpenFeatureAPI` 
+singleton to desired Spring components.
+
+```java
```

Comment:
Could you explicitly call out what the file should be named and where it should live?